core/commands/inline_diff.py

```diff
@@ -219,26 +219,33 @@ def get_object_contents(self, object_hash):
         """
         return self.git("show", "--no-color", object_hash)
 
+    def get_file_contents_binary(self, file_path):
+        """
+        Given an absolute file path, return the binary contents of that file
+        as a string.
+        """
+        file_path = os.path.join(self.repo_path, file_path)
+        with open(file_path, "rb") as f:
+            return f.read()
+
     def get_file_contents(self, file_path):
         """
         Given an absolute file path, return the text contents of that file
         as a string.
         """
-        file_path = os.path.join(self.repo_path, file_path)
+        binary = self.get_file_contents_binary(file_path)
         try:
-            with open(file_path, "rt", encoding="utf-8") as f:
-                return f.read()
+            return binary.decode('utf-8')
         except UnicodeDecodeError as unicode_err:
-            with open(file_path, "rt", encoding="latin-1") as f:
-                return f.read()
+            return binary.decode('latin-1')
```

Comment:
Is it the same with `codecs.decode()`? I ask because this is the last fallback in GitGutter if any other decoding failed. Wouldn't make sense if it defaults to utf-8, too.